metricbeat/module/dropwizard/collector/data.go

```diff
@@ -107,12 +111,19 @@ func splitTagsFromMetricName(metricName string) (string, common.MapStr) {
 		dobreak := false
 		ind := strings.Index(tagStr, ",")
 		eqPos := strings.Index(tagStr, "=")
+		if eqPos == -1 {
+			return metricName, nil
+		}
 		if ind == -1 {
 			tags[tagStr[:eqPos]] = tagStr[eqPos+1:]
 			dobreak = true
 		} else {
 			tags[tagStr[:eqPos]] = tagStr[eqPos+1 : ind]
-			tagStr = tagStr[ind+2:]
+			if ind+2 < len(tagStr) {
+				tagStr = tagStr[ind+2:]
```

Comment:
Why is it `+2`? comma and space? Is one and only one space mandatory?

Reply:
The planned feature from dropwizard is based on Java's Map toString() method, which adds one space after the comma. I updated the code to make the module more liberal in what it accepts.